x/wasm/internal/keeper/keeper.go

```diff
@@ -496,12 +510,26 @@ func (k Keeper) peekAutoIncrementID(ctx sdk.Context, lastIDKey []byte) uint64 {
 	return id
 }
 
-func (k Keeper) setAutoIncrementID(ctx sdk.Context, lastIDKey []byte, val uint64) {
+func (k Keeper) importAutoIncrementID(ctx sdk.Context, lastIDKey []byte, val uint64) {
 	store := ctx.KVStore(k.storeKey)
+	if store.Has(lastIDKey) {
+		panic(fmt.Sprintf("duplicate autoincrement id: %s", string(lastIDKey)))
+	}
 	bz := sdk.Uint64ToBigEndian(val)
 	store.Set(lastIDKey, bz)
 }
 
+func (k Keeper) importContract(ctx sdk.Context, address sdk.AccAddress, c *types.ContractInfo, state []types.Model) {
+	if !k.containsCodeInfo(ctx, c.CodeID) {
+		panic(fmt.Sprintf("unknown code id: %d", c.CodeID))
```

Comment:
Welcome to the dark side :dagger: 

I knew the sdk's "panic before error" would get you one day :wink:

Reply:
Ja, haha 👿  and there was not even a reason to panic here. I knew the error could not be propagated beyond the sdk boundary but still no need to panic here. 💩  will fix it. Thanks